### PR TITLE
fix: iOS beagle view backgroundColor and server driven state

### PR DESCRIPTION
**Pull request by gabcoelho**
### Related Issues

Closes: #1286 

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift [hunk 1]
```diff
@@ -218,26 +218,38 @@ public class BeagleScreenViewController: BeagleController {
...
+            renderScreenIfNeeded(state: serverDrivenState)
             serverDrivenState = .success
-            renderScreenIfNeeded()
+            renderScreenIfNeeded(state: serverDrivenState)
         case .failure(let error):
-            renderScreenIfNeeded()
             serverDrivenState = .finished
+            renderScreenIfNeeded(state: serverDrivenState)
             serverDrivenState = .error(error, viewModel.loadScreen)
+            renderScreenIfNeeded(state: serverDrivenState)
```
Comment (victorrabelozup) [inline iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift:235]: may be improved

File: iOS/Sources/Beagle/Sources/Renderer/BeagleView.swift [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 import UIKit
 
+public typealias BeagleViewState = (ServerDrivenState, UIView) -> ()
```
Comment (botbeaglezup) [inline iOS/Sources/Beagle/Sources/Renderer/BeagleView.swift:19]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/e8b7b4e1f60d6aae75aa4bed312359dc35854de2/iOS/Sources/Beagle/Sources/Renderer/BeagleView.swift#L19"></span>Prefer <code>-&gt; Void</code> over <code>-&gt; ()</code>.<br />
<code>void_return</code> <code>BeagleView.swift:19</code></td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/e8b7b4e1f60d6aae75aa4bed312359dc35854de2/iOS/Sources/Beagle/Sources/Components/ServerDrivenComponent/TabBar/TabBarUIComponent.swift#L238">iOS/Sources/Beagle/Sources/Components/ServerDrivenComponent/TabBar/TabBarUIComponent.swift#L238</a> - TODOs should be resolved (setNeedLayout should call layo…).<br />
<code>todo</code> <code>TabBarUIComponent.swift:238</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/e8b7b4e1f60d6aae75aa4bed312359dc35854de2/iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift#L58">iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift#L58</a> - TODOs should be resolved (This workaround should be remo…).<br />
<code>todo</code> <code>BeagleScreenViewController.swift:58</code></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_ios">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by gabrielleonardozup — APPROVED** [review]

**Review by dantes-git — APPROVED** [review]

File: iOS/Sources/Beagle/Sources/Renderer/BeagleView.swift [hunk 3]
```diff
@@ -16,6 +16,8 @@
 
 import UIKit
 
+public typealias BeagleViewState = (ServerDrivenState) -> Void
```
Comment (theffc) [inline iOS/Sources/Beagle/Sources/Renderer/BeagleView.swift:19]: I feel like using  the name `BeagleViewState` is somewhat misleading for a closure that receives the state.
It seems that this closure should be used like an observer/handler/delegate, and so maybe we should name it like that.
Does this makes sense?

File: iOS/Example/BeagleDemo/BeagleDemo/Screens/NativeViewController.swift [hunk 4]
```diff
@@ -44,14 +44,37 @@ class NativeViewController: UIViewController {
...
+            self.errorLabel.text = serverDrivenError.localizedDescription
+            self.errorLabel.textColor = .red
+            self.errorLabel.isHidden = false
+        case .success:
+            self.loadingLabel.isHidden = true
+            self.errorLabel.isHidden = true
+        case .finished:
+            break
+        }
+    }
```
Comment (victorrabelozup) [inline iOS/Example/BeagleDemo/BeagleDemo/Screens/NativeViewController.swift:64]:
```suggestion
        case .started:
            self.loadingLabel.isHidden = false
            self.errorLabel.isHidden = true
        case .error(var serverDrivenError, let retry):
            self.errorLabel.text = serverDrivenError.localizedDescription
            self.errorLabel.textColor = .red
            self.errorLabel.isHidden = false
        case .success:
            self.errorLabel.isHidden = true
        case .finished:
            self.loadingLabel.isHidden = true
        }
    }
```

Comment (gabcoelho) [reply iOS/Example/BeagleDemo/BeagleDemo/Screens/NativeViewController.swift]: good catch 😁 done!

**Review by victorrabelozup — APPROVED** [review]

**Review by dantes-git — APPROVED** [review]

**Pull request merged**